### PR TITLE
ensure we make progress even in manual exec

### PR DESCRIPTION
**Pull request by RensR**
We want to make sure that no progress is ever being reverted. This means that manual exec will not revert if the tx was untouched

### CODE REVIEW

File: contracts/scripts/native_solc_compile_all_ccip [hunk 1]
```diff
@@ -8,7 +8,7 @@ echo " └───────────────────────
 
 SOLC_VERSION="0.8.24"
 OPTIMIZE_RUNS=26000
-OPTIMIZE_RUNS_OFFRAMP=19500
```
Comment (RensR) [inline contracts/scripts/native_solc_compile_all_ccip:11]: We'll golf at the final end, this is fine now

File: contracts/src/v0.8/ccip/offRamp/EVM2EVMOffRamp.sol [hunk 2]
```diff
@@ -427,10 +431,9 @@ contract EVM2EVMOffRamp is IAny2EVMOffRamp, AggregateRateLimiter, ITypeAndVersio
         // prepended by the 4 bytes of ReceiverError.selector, TokenHandlingError.selector or InvalidPoolAddress.selector.
         // Max length of revert data is Router.MAX_RET_BYTES, max length of err is 4 + Router.MAX_RET_BYTES
         return (Internal.MessageExecutionState.FAILURE, err);
-      } else {
```
Comment (RensR) [inline contracts/src/v0.8/ccip/offRamp/EVM2EVMOffRamp.sol:430]: No change, the else didn't actually do anything

File: contracts/src/v0.8/ccip/test/offRamp/EVM2EVMMultiOffRampSetup.t.sol [hunk 3]
```diff
@@ -326,7 +326,7 @@ contract EVM2EVMMultiOffRampSetup is TokenSetup, PriceRegistrySetup, MultiOCR3Ba
     return message;
   }
 
-  function _generateBasicMessages(
+  function _generateSingleBasicMessage(
```
Comment (RensR) [inline contracts/src/v0.8/ccip/test/offRamp/EVM2EVMMultiOffRampSetup.t.sol:329]: Wrongly implies multiple msgs, so renamed it

File: contracts/src/v0.8/ccip/test/offRamp/EVM2EVMMultiOffRamp.t.sol [hunk 4]
```diff
@@ -1587,7 +1590,33 @@ contract EVM2EVMMultiOffRamp_manuallyExecute is EVM2EVMMultiOffRampSetup {
...
+        EVM2EVMMultiOffRamp.ReceiverError.selector,
+        abi.encodeWithSelector(MaybeRevertMessageReceiver.CustomError.selector, "")
+      )
+    );
+
+    uint256[][] memory gasLimitOverrides = new uint256[][](1);
+    gasLimitOverrides[0] = _getGasLimitsFromMessages(messages);
+
+    s_offRamp.manuallyExecute(_generateBatchReportFromMessages(SOURCE_CHAIN_SELECTOR_1, messages), gasLimitOverrides);
+  }
```
Comment (RayXpub) [inline contracts/src/v0.8/ccip/test/offRamp/EVM2EVMMultiOffRamp.t.sol:1621]: suggestion: wdyt of adding a follow up execution for a new message to assert that the new tx doesn't revert?

**Review by RayXpub — APPROVED** [review]

**Review by connorwstein — APPROVED** [review]

**Review by matYang — APPROVED** [review]

**Review by elatoskinas — APPROVED** [review]

**Pull request merged**